### PR TITLE
Update Feast Core API to support Project namespacing

### PR DESCRIPTION
**Pull request by woop**
Updated Feast Core gRPC API based on: [Feast Projects RFC](https://docs.google.com/document/d/14-QBz9X8zK_aGAY2ti43a7PqMxvqp0035ec7QYgmEBM/edit#)

This will be one of many PRs to the `project-isolation` branch. I am purposefully not merging this into master (as that will break master) and not building one massive PR (which will be too hard to parse).

### CODE REVIEW

File: protos/feast/core/CoreService.proto [hunk 1]
```diff
@@ -112,6 +133,9 @@ message ListStoresResponse {
 }
 
 message ApplyFeatureSetRequest {
+    // Name of project that the feature set belongs to.
+    string project = 2;
```
Comment (khorshuheng) [inline protos/feast/core/CoreService.proto]: Will this becomes a mandatory parameter?

File: protos/feast/core/CoreService.proto [hunk 2]
```diff
@@ -153,4 +178,34 @@ message UpdateStoreResponse {
...
+    // Name of project (required)
+    string name = 1;
+}
+
+// Response for creation of a project
+message CreateProjectResponse {
+}
+
+// Request for the archival of a project
+message ArchiveProjectRequest {
```
Comment (khorshuheng) [inline protos/feast/core/CoreService.proto]: What happens to the features ingested when a project is archived?

File: protos/feast/core/CoreService.proto (hunk 1 above)
Comment (woop) [reply protos/feast/core/CoreService.proto]: Yes

File: protos/feast/core/CoreService.proto (hunk 2 above)
Comment (woop) [reply protos/feast/core/CoreService.proto]: It continues to work. Ingestion is unaffected.

**Comment by woop** [comment]
I just realized that we should probably change the test scripts to also code gen from protos. Right now these tests are passing but they shouldnt. cc @davidheryanto @zhilingc

**Comment by davidheryanto** [comment]
Should we have a spec for the project name? For example,
>  Project name can be up to 253 characters long. The characters allowed in names are: digits (0-9), lower case letters (a-z), -, and ..

**Comment by woop** [comment]
> Should we have a spec for the project name? [For example](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names),
> 
> > Project name can be up to 253 characters long. The characters allowed in names are: digits (0-9), lower case letters (a-z), -, and .
> 
> So that when we're combining project name with feature name, there won't be ambiguity (e.g. if we use colon as a separator and project name includes colon as well). And the storage system where we persist the project name information may only support certain character sets, e.g. ASCII but not Unicode. And if we plan to use the project name in some URI, probably it should be DNS friendly too?

Yip, makes sense. Where should this be documented?
